### PR TITLE
OCPBUGS-37663: azure: Fix empty route table being created.

### PR DESCRIPTION
**Pull request by rna-afk**
Adding virtual network ID value of the AzureCluster to the existing vnet specified during install config to avoid CAPI setting the "managed" tag to the resources resulting in an empty route table being created.

### CODE REVIEW

File: pkg/asset/manifests/azure/cluster.go [hunk 1]
```diff
@@ -91,6 +92,21 @@ func GenerateClusterAssets(installConfig *installconfig.InstallConfig, clusterID
...
+	virtualNetworkID := ""
+	if installConfig.Config.Azure.VirtualNetwork != "" {
+		client, err := installConfig.Azure.Client()
+		if err != nil {
+			return nil, fmt.Errorf("failed to get azure client: %w", err)
+		}
+
+		virtualNetwork, err := client.GetVirtualNetwork(context.TODO(), installConfig.Config.Azure.NetworkResourceGroupName, installConfig.Config.Azure.VirtualNetwork)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get azure client: %w", err)
```
Comment (sadasu) [inline pkg/asset/manifests/azure/cluster.go:104]: Copy paste error. I think the error message should be different here.

File: pkg/asset/manifests/azure/cluster.go [hunk 2]
```diff
@@ -91,6 +92,21 @@ func GenerateClusterAssets(installConfig *installconfig.InstallConfig, clusterID
...
+		}
+
```
Comment (sadasu) [inline pkg/asset/manifests/azure/cluster.go:106]: Even when no error is returned, maybe a good idea to check if virtualNetwork != nil before accessing the ID.

**Review by jhixson74 — APPROVED** [review]

**Comment by sadasu** [comment]
/override ci/prow/e2e-aws-ovn

Known issue with AWS jobs today. This change only affects Azure.

**Pull request merged**